src/realm/object-store/c_api/schema.cpp

```diff
@@ -54,6 +54,14 @@ RLM_API bool realm_schema_validate(const realm_schema_t* schema)
     });
 }
 
+RLM_API bool realm_update_schema(realm_t* realm, const realm_schema_t* schema)
+{
+    return wrap_err([&]() {
+        realm->get()->update_schema(*schema->ptr);
+        return true;
```

Comment:
I know this is inline with precedent, but wouldn't it make more sense to specialize `wrap_err` on void functions and have it return `true` for success and `false` for error? It seems a bit silly that we have to add the fake `return true` to all void functions.

Reply:
I'm not sure there is a way to do this without some serious overhead. We could change `wrap_err` to:
```
template <class F>
static inline auto wrap_err(F&& f) -> std::conditional_t<std::is_void_v<decltype(std::declval<F>()())>, bool, decltype(std::declval<F>()())>
{
    try {
        if constexpr (std::is_void_v<decltype(std::declval<F>()())>) {
            f();
            return true;
        } else {
            return f();
        }
    }
    catch (...) {
        set_last_exception(std::current_exception());
        if constexpr (std::is_void_v<decltype(std::declval<F>()())>) {
            return false;
        } else {
            return {};
        }
    };
}
```
But it's a nasty bit of magic that I don't think would be appreciated. Maybe @simonask has some darker magic he could conjure to avoid type traits, but I'm not sure it's worth it.

Reply:
Blah... one would imagine that a language as advanced as C++ would have a nicer way to overload a function on the return type of its argument 🙄

Reply:
C++ has a pretty dumb type system, which fundamentally prohibits overloading on the return type (at least without unspeakable acts of dark sorcery).

But I don't understand why you would say that `return true` is fake here? The function is meant to return true when it succeeds, so that is perfectly fine? It's not a void function.

Reply:
The lambda that updates the schema is (or at least can be) a void function. The boolean return type of the parent function is an artifact of the error handling strategy that we chose. If C++ had a nicer way to express overloads on the argument return type, this could be rewritten as

```
return wrap_err([&]() {
    realm->get()->update_schema(*schema->ptr);
});
```

With no change in behavior and no information being lost.